### PR TITLE
Multi rfq send itest

### PR DESCRIPTION
**Pull request by GeorgeTsagk**
## Description

Enhances the multi-rfq itest to cover multi-rfq send functionality. In the following topology
```
	  /---[sats]--> Erin --[assets]--\
	 /                                \
	/                                  \

 Charlie  -----[sats]-->  Dave  --[assets]---->Fabia

	\                                  /
	 \                                /
	  \---[sats]--> Yara --[assets]--/
```

we now also get Fabia to pay Charlie back, with amounts that exceed the capacity of each individual channel. We also use a hold invoice to validate that multiple HTLCs were added in different channels.

Related PRs (in order of merge sequence):

- [x] https://github.com/lightningnetwork/lnd/pull/9980
- [ ] https://github.com/lightninglabs/taproot-assets/pull/1613
- [ ] https://github.com/lightninglabs/loop/pull/968

### CODE REVIEW

File: itest/assets_test.go [hunk 1]
```diff
@@ -1733,8 +1737,8 @@ func payInvoiceWithAssets(t *testing.T, payer, rfqPeer *HarnessNode,
 		acceptedQuote := quoteMsg.GetAcceptedSellOrder()
 		require.NotNil(t, acceptedQuote)
 
-		peerPubKey := acceptedQuote.Peer
-		require.Equal(t, peerPubKey, rfqPeer.PubKeyStr)
+		// peerPubKey := acceptedQuote.Peer
```
Comment (guggero) [inline itest/assets_test.go]: nit: commented out code.

File: itest/assets_test.go [hunk 2]
```diff
@@ -264,6 +307,19 @@ func createTestMultiRFQAssetNetwork(t *harnessTest, net *NetworkHarness,
 	require.NoError(t.t, err)
 	t.Logf("Funded channel between Yara and Fabia: %v", fundRespYF)
 
+	// We fund the Yara->Fabia channel.
```
Comment (guggero) [inline itest/assets_test.go]: nit: comment mismatch.

File: itest/litd_custom_channels_test.go [hunk 3]
```diff
@@ -94,7 +94,30 @@ var (
...
+		"--taproot-assets.universerpccourier.skipinitdelay",
+		"--taproot-assets.universerpccourier.backoffresetwait=1s",
+		"--taproot-assets.universerpccourier.numtries=5",
+		"--taproot-assets.universerpccourier.initialbackoff=300ms",
+		"--taproot-assets.universerpccourier.maxbackoff=600ms",
+		"--taproot-assets.universerpccourier.skipinitdelay",
+		"--taproot-assets.universerpccourier.backoffresetwait=100ms",
+		"--taproot-assets.universerpccourier.initialbackoff=300ms",
+		"--taproot-assets.universerpccourier.maxbackoff=600ms",
+		"--taproot-assets.custodianproofretrievaldelay=500ms",
```
Comment (guggero) [inline itest/litd_custom_channels_test.go]: nit: can use `append(litdArgsTemplateNoOracle, []string{` instead of duplicating all these values (since they look identical at first glance).

**Review by guggero — APPROVED** [review]
LGTM pending a couple of nits.

File: itest/litd_custom_channels_test.go [hunk 4]
```diff
@@ -3100,13 +3138,88 @@ func testCustomChannelsMultiRFQReceive(ctx context.Context, net *NetworkHarness,
...
+	require.NoError(t.t, err)
+
+	assertNumHtlcs(t.t, charlie, 0)
+	assertNumHtlcs(t.t, fabia, 0)
+
+	logBalance(t.t, nodes, assetID, "after multi-rfq send")
+
+	// Let's make another round-trip involving multi-rfq functionality.
+	// Let's have Fabia receive another large payment and send it back
+	// again, this time with a greater amount.
```
Comment (ZZiigguurraatt) [inline itest/litd_custom_channels_test.go:3187]: and you are also using a group key too instead of an asset id? does/should this test ensure that it shards and multiple asset ID are used in the group key or is it not trying to check for that?

File: go.mod [hunk 5]
```diff
@@ -247,3 +247,7 @@ replace google.golang.org/protobuf => github.com/lightninglabs/protobuf-go-hex-d
 // it is a replace in the tapd repository, it doesn't get propagated here
 // automatically, so we need to add it manually.
 replace github.com/golang-migrate/migrate/v4 => github.com/lightninglabs/migrate/v4 v4.18.2-9023d66a-fork-pr-2
+
```
Comment (ZZiigguurraatt) [inline go.mod]: please remove these lines now that https://github.com/lightningnetwork/lnd/pull/9980 has been merged

**Comment by ZZiigguurraatt** [comment]
> Need to remove the `replace` in `go.mod`, and some other outstanding comments, otherwise good to go.

FYI: I made a branch where I rebased and then updated all the packages. Not sure if you want to use it or re-do it another way.

https://github.com/ZZiigguurraatt/taproot-assets/tree/group-key-addr-part-2%2Bmulti-rfq-send

File: itest/litd_custom_channels_test.go (hunk 4 above)
Comment (GeorgeTsagk) [reply itest/litd_custom_channels_test.go:3187]: we do more extensive tests around assetID/groupkey in diff itests (mostly the liquidity edge cases) here I'm just switching things up a bit just as an extra sanity check

**Comment by guggero** [comment]
That loop PR doesn't have to be merged, since we're merging into a temporary side branch here as well (to which the `tapd` CI currently points).
